### PR TITLE
Adds q2

### PR DESCRIPTION
**Pull request by kristina-altunina**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        response.body.forEach((topic) => {
+            expect(topic).toHaveProperty('description', expect.any(String));
+            expect(topic).toHaveProperty('slug', expect.any(String));
+        });
+        });
+    });
+
+    it('200: responds with a single topic with slug', () => {
+        return request(app)
+        .get('/api/topics/cats')
```
Comment (Augs0) [inline __tests__/api.test.js]: This is not one of the endpoints. We only need to see the list of topics, but we will come back to filtering by topic in a much later task. Please remove this.

File: __tests__/api.test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    });
+
+    it('404: responds with an error message if the topic is doesn\'t exist', () => {
+        return request(app).
+        get('/api/topics/abc')
+        .then((response) => {
+            const {message} = response.body;
+            expect(message).toBe('404: Not Found')
+        });
+    });
```
Comment (Augs0) [inline __tests__/api.test.js]: We don't need a 400 test here as the endpoint is not supposed to take any queries and what you are testing is a different endpoint anyway (/api/topics/:topic). Please remove these tests too. The only error handler we could have here is a generic 404 for any paths that do not exist in our app. You could use your current 404 test here for that.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const express = require('express');
+const { getAllTopics, getTopicBySlug } = require('./controllers/topics-controller');
+const app = express();
+
+app.use('/api/topics/:slug', getTopicBySlug);
```
Comment (Augs0) [inline app.js]: This is not asked of you in the instructions. Can I ask why you chose to include this endpoint?

File: controllers/topics-controller.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    })
+    .catch((err) => {
+        next(err);
+    });
+};
+
+const getTopicBySlug = (request, response, next) => {
+    const {slug} = request.params;
+    if(!isNaN(slug)){
+        throw {status: 400};
```
Comment (Augs0) [inline controllers/topics-controller.js]: You are not throwing an error here. But, for what it is worth, a throw is not as good as a Promise.reject here for two reason. 1. it only allows us to pass a string so actually this object would be totally ignored and the status, error etc will all come from the Error thrown by the server or Express, 2.  throwing inside a Promise triggers a reject but it won't be passed to the catch in our promise chain and then to app; the error will just throw immediately so we can't customise what it does

File: controllers/topics-controller.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        }
+        response.status(200).send(topics);   
+    })
+    .catch((err) => {
+        next(err);
+    });
+};
+
+const getTopicByQuery = (request, response, next) => {
+    const {order_by} = request.params
```
Comment (Augs0) [inline controllers/topics-controller.js]: As mentioned previously, no queries are expected at this stage or on this endpoint

File: models/topics-model.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+const connection = require('../db/connection');
+const { topicData } = require('../db/data/test-data');
+
+const selectTopics = () => {
+    return connection.query(`SELECT * FROM topics`)
+                    .then(({rows : topics}) => {
```
Comment (Augs0) [inline models/topics-model.js]: What do we mean by rows : topics here? I also noticed you have an OR operator to return an empty array but PSQL would return that anyway if it couldn't find any records

**Review by Augs0 — CHANGES_REQUESTED** [review]
Hey Kristina, a good effort to start but some confusing bits of code here too. Please remove all the code relating to queries as we have not asked for these. Your model in particular is a little bit muddled up. If you need any help implementing the changes requested, feel free to put in a helpdesk request

**Comment by kristina-altunina** [comment]
Hi @Augs0, please see the changes I've made.
Thanks

**Review by B7Bprog — APPROVED** [review]
Hey Kristina, nice refactoring, please merge it. 🎉

**Pull request merged**